maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -816,6 +817,15 @@ private Model readEffectiveModel(
             // profile activation
             profileActivationContext.setProjectProperties(modelv4.getProperties());
 
+            // profile check
+            HashSet<String> profileIds = new HashSet<>();
```

Comment:
Can't this be done in the ModelValidator ?

Reply:
Should a `validateProfiles` method be added to the `ModelValidator` ?

Reply:
Validator does validate profiles. But for these "far" POMs we use consequently "min" level, that does NOT. 
In Mvn3 the idea was to detect this duplication while processing (as profiles do affect build a lot, can add extra properties, dependencies, and failure to properly evaluate them can lead to mysterious failures).
We can move ID validation to Min level, yes, as it currently validate very few things.

Reply:
I like these changes (having _validation in validator_), so am tempted to introduce this change in 3.9 line as well...